src/react/portfolio/VanillaJSProjects.tsx

```diff
@@ -1,116 +1,203 @@
+    githubLink: 'https://github.com/Navleenkaursandhu/notebook/tree/main/src/vanilla_js/tic_tac_toe'
   },
   {
-    title: 'FIGHTING GAME',
-    description: 'Fighting Game',
+    id: 'vanillajs-fightinggame',
+    title: 'STREET FIGHTING GAME',
+    description: <div>
+      Want to play something exciting? Try this 1-on-1 fighting game with your friend on your computer using your keyboard.
+      Please note that in order to play this game you need to press your keyboard alphabets to take actions during fight. The rules
+      are pretty simple. The game has 3 rounds in total. To start the game Player A takes the first round followed by player B and so forth. It will be the player&apos;s choice whether
+      he/she wants to hit the opponent or take a lifeline during his/her round (remember there is no point of taking
+      a lifeline in your first round as it&apos;s already 100%). On hit, opponents health will reduce by a random number (the damage will be reflected by red bar)
+      and on selecting lifeline the current player&apos;s health increases by random number (health is reflected by green bar). After
+      3rd round the player with good health wins the battle.
+    </div>,
     skills: ['JavaScript', 'CSS', 'HTML', 'Git', 'Github', 'VS Code'],
     date: 'July 2022 - August 2022',
     link: '/src/vanilla_js/fighting_game/game.html',
-    githubLink: 'https://github.com/Navleenkaursandhu/notebook/tree/main/projects/fighting_game'
+    githubLink: 'https://github.com/Navleenkaursandhu/notebook/tree/main/src/vanilla_js/fighting_game'
   },
   {
+    id: 'vanillajs-rpsgame',
     title: 'ROCK PAPER SCISSORS GAME',
-    description: 'Rock Paper Scissors Game',
+    description: <div>
+      The game is implemented using vanillaJS, HTML and CSS. Click on &quot;view&quot; button below to start playing and for instructions and rules you can
+      visit <a className='text-blue-600 underline underline-offset-2' href={`#${reactRPSGameID}`}>here</a>. For react App and code for this game,
+      visit react projects segment above on this page.
+    </div>,
     skills: ['JavaScript', 'CSS', 'HTML', 'Git', 'Github', 'VS Code'],
     date: 'July 2022',
     link: '/src/vanilla_js/rock_paper_scissors/rps.html',
-    githubLink: 'https://github.com/Navleenkaursandhu/notebook/tree/main/projects/rock_paper_scissors'
+    githubLink: 'https://github.com/Navleenkaursandhu/notebook/tree/main/src/vanilla_js/rock_paper_scissors'
   },
   {
+    id: 'vanillajs-tipcalculator',
     title: 'TIP CALCULATOR',
-    description: 'Tip Calculator',
+    description: <div>
+      Get the full description of this project <a className='text-blue-600 underline underline-offset-2' href={`#${reactTipCalculatorID}`}>here</a>. The JS code
+      and app can be visited using buttons below and you will be able to locate react code and app under react projects section above.
+    </div>,
```

Comment:
I see that you had to repeat similar text multiple times. But, each time you have different words for it. I suggest that you make a common component to handle this. You could define the component at the top of this file just after the imports. Or, you could define it in a separate file (probably the better thing to do). Maybe call it `ReactReimplementationRedirect`

In it, you could take the `href` as a prop and have text like this

```tsx
<div>
  This project has been re-implemented in React. You can see the React implementation and the full description
  of this project <a className='text-blue-600 underline underline-offset-2' href={props.href}>here</a>. You
  can still view the Vanilla JS implementation of the project and its source code below.
</div>
```

Then use it like this in all the projects you have to show this text like this

```suggestion
    description: <div>
      <ReactReimplementationRedirect href={`#${reactTipCalculatorID}`}/>
    </div>,
```

Reply:
done